columnflow/production/cms/seeds.py

```diff
@@ -119,15 +132,15 @@ def deterministic_jet_seeds(self: Producer, events: ak.Array, **kwargs) -> ak.Ar
     events = self[deterministic_event_seeds](events, **kwargs)
 
     # create the per jet seeds
-    prime_offset = 18
+    primes = self.primes[events.deterministic_seed % len(self.primes)]
     jet_seed = events.deterministic_seed + (
-        primes[prime_offset] * ak.values_astype(ak.local_index(events.Jet, axis=1), np.uint64)
+        primes * ak.values_astype(ak.local_index(events.Jet, axis=1) + 1, np.uint64)
```

Comment:
@Bogdan-Wiederspan Thinking about this again, we use the event seed to infer jet seeds using the local index. However, if we use the same idea for other objects (eg. leptons), per event we end up with the same seed for the objects (jet 0 will have the same seed as lepton 0, etc).

We should introduce an additional offset here and we should use a different one for future objects. I'd suggest the 50th prime rather than just `1`.

```suggestion
        primes * ak.values_astype(ak.local_index(events.Jet, axis=1) + self.primes[50], np.uint64)
```